server/main.lua

```diff
@@ -37,32 +37,70 @@ exports('DoesPlayerVehiclePlateExist', doesEntityPlateExist)
 ---@field id number
 ---@field citizenid? string
 ---@field modelName string
+---@field garage string
+---@field state State
+---@field depotPrice integer
 ---@field props table ox_lib properties table
 
----@class GetPlayerVehiclesRequest
+---@class PlayerVehiclesFilters
 ---@field vehicleId? number
----@field citizenid? string
----@field license? string
----@field plate? string
-
----@param idType IdType
----@param idValue string
----@return PlayerVehicle[]?, ErrorResult? errorResult
-local function getPlayerVehicles(idType, idValue)
-    local err = validateIdType(idType)
-    if err then return nil, err end
+---@field citizenId? string
+---@field states? State|State[]
+---@field garage? string
+
+---@param filters? PlayerVehiclesFilters
+---@return string whereClause, any[] placeholders
+local function buildWhereClause(filters)
+    if not filters then
+        return '', {}
+    end
+    local query = ' WHERE 1=1'
+    local placeholders = {}
+    if filters.vehicleId then
+        query = query .. ' AND id = ?'
+        placeholders[#placeholders+1] = filters.vehicleId
+    end
+    if filters.citizenId then
+        query = query .. ' AND citizenid = ?'
+        placeholders[#placeholders+1] = filters.citizenId
+    end
+    if filters.garage then
+        query = query .. ' AND garage = ?'
+        placeholders[#placeholders+1] = filters.garage
+    end
+    if filters.states then
+        if type(filters.states) ~= 'table' then
+            ---@diagnostic disable-next-line: assign-type-mismatch
+            filters.states = {filters.states}
+        end
+        if #filters.states > 0 then
+            query = query .. ' AND (1=2'
+            for i = 1, #filters.states do
+                local state = filters.states[i]
+                query = query .. ' OR state = ?'
+                placeholders[#placeholders+1] = state
+            end
+            query = query .. ')'
+        end
+    end
```

Comment:
Little weird to read but I guess when you are programmatically assembling SQL statements it's gonna be weird.

Reply:
I improved the states readability a bit using table.concat. It sounds like Artur has an idea for further improvement. Will resolve for now, maybe this can be improved in a further PR.